sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs

```diff
@@ -274,9 +387,70 @@ private async ValueTask<Response> GetResponseAsync(bool async, string uri, Cance
             {
                 _pipeline.Send(message, cancellationToken);
             }
+
+            // If we are doing final get for a delete LRO with 404, just return empty response with 204
+            if (message.Response.Status == 404 && RequestMethod == RequestMethod.Delete)
+            {
+                return new EmptyResponse(HttpStatusCode.NoContent);
+            }
             return message.Response;
         }
 
+        /// <summary>
+        /// This is only used for final get of the delete LRO, we just want to return an empty response with 204 to the user for this case.
+        /// </summary>
+        private sealed class EmptyResponse : Response
+        {
+            public EmptyResponse(HttpStatusCode status)
+            {
+                Status = (int)status;
+                ReasonPhrase = status.ToString();
+            }
+
+            public override int Status { get; }
+
+            public override string ReasonPhrase { get; }
+
+            public override Stream? ContentStream { get => null; set => throw new System.NotImplementedException(); }
```

Comment:
I think it is discouraged to return NotImplementedException.  Is there another exception type that might help the end-user understand what went wrong?  It's also recommended provide an exception message that 1) explains what went wrong 2) helps the user understand how to fix it.

Reply:
Updated to 
```csharp
public override Stream? ContentStream { get => null; set => throw new InvalidOperationException("Should not set ContentStream for an empty response."); }
```